When using Terraform CLI with command show we get a new error: TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined

**Describe the bug**
We are using the Terraform CLI DevOps extension version 0.6.8.
This always worked and now it has started to fail with minimal error information.

**To Reproduce**
Steps to reproduce the behavior:
1. Using the extension run a **terraform plan** that outputs the plan to a file
```
- task: TerraformCLI@0
      displayName: 'Terraform Plan'
      condition: and(
          always(),
          succeeded()
        )
      inputs:
        command: 'plan'
        commandOptions: '-lock-timeout="${{ parameters.terraform_lock_timeout }}" -out plan.tfplan -detailed-exitcode -input=false'
        environmentServiceName: '${{ parameters.deploymentServiceConnection }}'
        workingDirectory: '$(Build.SourcesDirectory)/automation/terraform/${{ parameters.tf_module }}'
      env:
        TF_VAR_config_root: $(Build.SourcesDirectory)/${{ parameters.configurationPath }}
        TF_VAR_config_directory: ${{ parameters.config }}
        ${{ each tf_var in parameters.tf_vars }}:
          TF_VAR_${{ tf_var.Key }}: ${{ tf_var.Value }}
        ${{ each env_var in parameters.env_vars }}:
          ${{ env_var.Key }}: ${{ env_var.Value }}
```
          
2. Using the extension run a **terraform show** with the plan file from step 1
```
    - task: TerraformCLI@0
      displayName: 'Detect Destroy Actions'
      condition: and(
          always(),
          succeeded(),
          eq(variables['TERRAFORM_PLAN_HAS_CHANGES'], 'true')
        )
      inputs:
        command: 'show'
        inputTargetPlanOrStateFilePath: 'plan.tfplan'
        workingDirectory: '$(Build.SourcesDirectory)/automation/terraform/${{ parameters.tf_module }}'
```

3. See error:
```
##[error]TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
```

**Expected behavior**
We have been using this for a while and we use it to detect if anything in the plan is going to be detroyed

**Screenshots**
![image](https://user-images.githubusercontent.com/1108859/117937015-e2dbb980-b348-11eb-93d2-8c797e8f53ac.png)

**Pipeline Logs**
2021-05-12T03:12:53.9955373Z ##[debug]Evaluating condition for step: 'Detect Destroy Actions'
2021-05-12T03:12:53.9960260Z ##[debug]Evaluating: and(always(), succeeded(), eq(variables['TERRAFORM_PLAN_HAS_CHANGES'], 'true'))
2021-05-12T03:12:53.9961993Z ##[debug]Evaluating and:
2021-05-12T03:12:53.9963407Z ##[debug]..Evaluating always:
2021-05-12T03:12:53.9965115Z ##[debug]..=> True
2021-05-12T03:12:53.9966112Z ##[debug]..Evaluating succeeded:
2021-05-12T03:12:53.9967676Z ##[debug]..=> True
2021-05-12T03:12:53.9968638Z ##[debug]..Evaluating eq:
2021-05-12T03:12:53.9969631Z ##[debug]....Evaluating indexer:
2021-05-12T03:12:53.9970778Z ##[debug]......Evaluating variables:
2021-05-12T03:12:53.9972229Z ##[debug]......=> Object
2021-05-12T03:12:53.9973855Z ##[debug]......Evaluating String:
2021-05-12T03:12:53.9976407Z ##[debug]......=> 'TERRAFORM_PLAN_HAS_CHANGES'
2021-05-12T03:12:53.9978350Z ##[debug]....=> 'true'
2021-05-12T03:12:53.9980048Z ##[debug]....Evaluating String:
2021-05-12T03:12:53.9981986Z ##[debug]....=> 'true'
2021-05-12T03:12:53.9983188Z ##[debug]..=> True
2021-05-12T03:12:53.9984393Z ##[debug]=> True
2021-05-12T03:12:53.9986328Z ##[debug]Expanded: and(True, True, eq('true', 'true'))
2021-05-12T03:12:53.9987588Z ##[debug]Result: True
2021-05-12T03:12:53.9988998Z ##[section]Starting: Detect Destroy Actions
2021-05-12T03:12:54.0006679Z ==============================================================================
2021-05-12T03:12:54.0007671Z Task         : Terraform CLI
2021-05-12T03:12:54.0008417Z Description  : Execute terraform cli commands
2021-05-12T03:12:54.0009757Z Version      : 0.6.8
2021-05-12T03:12:54.0010847Z Author       : Charles Zipp
2021-05-12T03:12:54.0011391Z Help         : 
2021-05-12T03:12:54.0011991Z ==============================================================================
2021-05-12T03:12:54.0301461Z ##[debug]Using node path: /apps/agent/externals/node/bin/node
2021-05-12T03:12:54.2904532Z ##[debug]agent.TempDirectory=/apps/agent/_work/_temp
2021-05-12T03:12:54.3125178Z ##[debug]loading inputs and endpoints
2021-05-12T03:12:54.3142834Z ##[debug]loading INPUT_COMMAND
2021-05-12T03:12:54.3215268Z ##[debug]loading INPUT_WORKINGDIRECTORY
2021-05-12T03:12:54.3222509Z ##[debug]loading INPUT_RUNAZLOGIN
2021-05-12T03:12:54.3229575Z ##[debug]loading INPUT_BACKENDTYPE
2021-05-12T03:12:54.3231393Z ##[debug]loading INPUT_ENSUREBACKEND
2021-05-12T03:12:54.3233031Z ##[debug]loading INPUT_BACKENDAZURERMSTORAGEACCOUNTSKU
2021-05-12T03:12:54.3234915Z ##[debug]loading INPUT_AIINSTRUMENTATIONKEY
2021-05-12T03:12:54.3236539Z ##[debug]loading INPUT_ALLOWTELEMETRYCOLLECTION
2021-05-12T03:12:54.3238664Z ##[debug]loading INPUT_INPUTTARGETPLANORSTATEFILEPATH
2021-05-12T03:12:54.3240287Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2021-05-12T03:12:54.3241920Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2021-05-12T03:12:54.3244221Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2021-05-12T03:12:54.3246917Z ##[debug]loading SECRET_xxxxxxxxx-AZURERM-SPN-CLIENT-ID
2021-05-12T03:12:54.3249137Z ##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
2021-05-12T03:12:54.3251620Z ##[debug]loading SECRET_xxxxxxxxx-AZUREAD-SPN-CLIENT-ID
2021-05-12T03:12:54.3254298Z ##[debug]loading SECRET_xxxxxxxxx-AZURERM-SPN-SECRET
2021-05-12T03:12:54.3256126Z ##[debug]loading SECRET_xxxxxxxxxAZDOAUTOMATION-PAT-MODULE-DOWNLOAD
2021-05-12T03:12:54.3257776Z ##[debug]loading SECRET_AZDO_PERSONAL_ACCESS_TOKEN
2021-05-12T03:12:54.3259719Z ##[debug]loading SECRET_xxxxxxxxx-AZUREAD-SPN-SECRET
2021-05-12T03:12:54.3262164Z ##[debug]loading SECRET_xxxxxxxxxAZDOAUTOMATION-PAT
2021-05-12T03:12:54.3265743Z ##[debug]loading SECRET_xxxxxxxxx-SONARQUBE-TOKEN
2021-05-12T03:12:54.3268467Z ##[debug]loading SECRET_xxxxxxxxx-PIMINTERFACE-SECRET
2021-05-12T03:12:54.3270490Z ##[debug]loading SECRET_xxxxxxxxx-NETIQ-IDM-INTEGRATION-PARTNER-USERNAME
2021-05-12T03:12:54.3272233Z ##[debug]loading SECRET_ARTIFACTORY_PROVIDER_API_KEY
2021-05-12T03:12:54.3274462Z ##[debug]loading SECRET_xxxxxxxxx-ARTIFACTORY-ACCESS-TOKEN
2021-05-12T03:12:54.3276819Z ##[debug]loading SECRET_xxxxxxxxx-ARTIFACTORY-TOKEN-NAME
2021-05-12T03:12:54.3279244Z ##[debug]loading SECRET_xxxxxxxxx-NETIQ-IDM-RESTDRIVER-USERNAME
2021-05-12T03:12:54.3281284Z ##[debug]loaded 27
2021-05-12T03:12:54.3283021Z ##[debug]Agent.ProxyUrl=undefined
2021-05-12T03:12:54.3284808Z ##[debug]Agent.CAInfo=undefined
2021-05-12T03:12:54.3286550Z ##[debug]Agent.ClientCert=undefined
2021-05-12T03:12:54.3288459Z ##[debug]Agent.SkipCertValidation=undefined
2021-05-12T03:12:54.7754827Z ##[debug]allowTelemetryCollection=true
2021-05-12T03:12:54.7758749Z ##[debug]aiInstrumentationKey=ec11a02f-c97a-4b55-8005-acf129682668
2021-05-12T03:12:54.8100004Z ##[debug]System.TeamFoundationCollectionUri=https://dev.azure.com/xxxxgroup/
2021-05-12T03:12:54.8106019Z ##[debug]System.TeamProject=xxxxxxxxx
2021-05-12T03:12:54.8108534Z ##[debug]System.HostType=build
2021-05-12T03:12:54.8110301Z ##[debug]Agent.OS=Linux
2021-05-12T03:12:54.8111975Z ##[debug]Agent.OSArchitecture=X64
2021-05-12T03:12:54.8116591Z ##[debug]Agent.JobStatus=Succeeded
2021-05-12T03:12:54.8122326Z ##[debug]System.TeamFoundationCollectionUri=https://dev.azure.com/xxxxgroup/
2021-05-12T03:12:54.8129911Z ##[debug]SYSTEMVSSCONNECTION auth param ACCESSTOKEN = ***
2021-05-12T03:12:54.8131693Z ##[debug]Agent.ProxyUrl=undefined
2021-05-12T03:12:54.8171722Z ##[debug]command=show
2021-05-12T03:12:54.8176400Z ##[debug]command=show
2021-05-12T03:12:54.8184136Z ##[debug]workingDirectory=/apps/agent/_work/8/s/automation/terraform/deploy_devops_tools
2021-05-12T03:12:54.8195031Z ##[debug]which 'terraform'
2021-05-12T03:12:54.8201930Z ##[debug]found: '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:54.8204857Z ##[debug]which '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:54.8207908Z ##[debug]found: '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:54.8212803Z ##[debug]/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform arg: version
2021-05-12T03:12:54.8223976Z ##[debug]exec tool: /apps/agent/_work/_tool/terraform/0.13.4/x64/terraform
2021-05-12T03:12:54.8226859Z ##[debug]arguments:
2021-05-12T03:12:54.8228865Z ##[debug]   version
2021-05-12T03:12:54.8236041Z [command]/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform version
2021-05-12T03:12:54.8889183Z (node:17787) Warning: Ignoring extra certs from `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`, load failed: error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table
2021-05-12T03:12:54.8891135Z 
2021-05-12T03:12:55.4954243Z Terraform v0.13.4
2021-05-12T03:12:55.4955835Z + provider registry.terraform.io/hashicorp/azuread v1.4.0
2021-05-12T03:12:55.4957099Z + provider registry.terraform.io/hashicorp/null v3.1.0
2021-05-12T03:12:55.4958790Z + provider registry.terraform.io/hashicorp/random v3.0.0
2021-05-12T03:12:55.4959899Z + provider registry.terraform.io/hashicorp/time v0.7.1
2021-05-12T03:12:55.4960943Z + provider registry.terraform.io/microsoft/azuredevops v0.1.2
2021-05-12T03:12:55.4962285Z + provider tfe.cmltd.net.au/xxxx/artifactory v0.0.14+27
2021-05-12T03:12:55.4963363Z + provider tfe.cmltd.net.au/xxxx/sonarqube v0.0.38+53
2021-05-12T03:12:55.4964263Z 
2021-05-12T03:12:55.4965172Z Your version of Terraform is out of date! The latest version
2021-05-12T03:12:55.4966797Z is 0.15.3. You can update by downloading from https://www.terraform.io/downloads.html
2021-05-12T03:12:55.5002522Z 
2021-05-12T03:12:55.5012477Z ##[debug]Exit code 0 received from tool '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:55.5016198Z ##[debug]STDIO streams have closed for tool '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:55.5049960Z ##[debug]secureVarsFile=undefined
2021-05-12T03:12:55.5051841Z ##[debug]secureVarsFile=undefined
2021-05-12T03:12:55.5053904Z ##[debug]secure file name for id undefined = undefined
2021-05-12T03:12:55.5056060Z ##[debug]commandOptions=undefined
2021-05-12T03:12:55.5057738Z ##[debug]commandOptions=undefined
2021-05-12T03:12:55.5060609Z ##[debug]inputTargetPlanOrStateFilePath=plan.tfplan
2021-05-12T03:12:55.5075898Z ##[debug]command=show
2021-05-12T03:12:55.5077972Z ##[debug]workingDirectory=/apps/agent/_work/8/s/automation/terraform/deploy_devops_tools
2021-05-12T03:12:55.5095492Z ##[debug]which 'terraform'
2021-05-12T03:12:55.5098293Z ##[debug]found: '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:55.5101737Z ##[debug]which '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:55.5104901Z ##[debug]found: '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:55.5107100Z ##[debug]/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform arg: show
2021-05-12T03:12:55.5109608Z ##[debug]/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform arg: -json
2021-05-12T03:12:55.5112279Z ##[debug]/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform arg: plan.tfplan
2021-05-12T03:12:55.5114390Z ##[debug]exec tool: /apps/agent/_work/_tool/terraform/0.13.4/x64/terraform
2021-05-12T03:12:55.5116263Z ##[debug]arguments:
2021-05-12T03:12:55.5117713Z ##[debug]   show
2021-05-12T03:12:55.5119277Z ##[debug]   -json
2021-05-12T03:12:55.5120772Z ##[debug]   plan.tfplan
2021-05-12T03:12:59.7406299Z ##[debug]Exit code 0 received from tool '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:59.7570013Z ##[debug]STDIO streams have closed for tool '/apps/agent/_work/_tool/terraform/0.13.4/x64/terraform'
2021-05-12T03:12:59.7655288Z ##[debug]inputTargetPlanOrStateFilePath=plan.tfplan
2021-05-12T03:12:59.7765094Z ##[debug]inputTargetPlanOrStateFilePath=plan.tfplan
2021-05-12T03:12:59.7767086Z ##[debug]allowTelemetryCollection=true
2021-05-12T03:12:59.7825239Z ##[error]TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
2021-05-12T03:12:59.7839370Z ##[debug]Processed: ##vso[task.issue type=error;]TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
2021-05-12T03:12:59.7841067Z ##[debug]allowTelemetryCollection=true
2021-05-12T03:12:59.7848498Z ##[debug]command=show
2021-05-12T03:12:59.7849889Z ##[debug]command=show
2021-05-12T03:12:59.7851343Z ##[debug]command=show
2021-05-12T03:12:59.7853119Z ##[debug]executed command 'show'
2021-05-12T03:12:59.7855961Z ##[debug]set TERRAFORM_LAST_EXITCODE=1
2021-05-12T03:12:59.7861338Z ##[debug]Processed: ##vso[task.setvariable variable=TERRAFORM_LAST_EXITCODE;isOutput=false;issecret=false;]1
2021-05-12T03:12:59.7862815Z ##[debug]task result: Failed
2021-05-12T03:12:59.7865628Z ##[error]TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
2021-05-12T03:12:59.7868440Z ##[debug]Processed: ##vso[task.issue type=error;]TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
2021-05-12T03:12:59.7872780Z ##[debug]Processed: ##vso[task.complete result=Failed;]TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
2021-05-12T03:13:00.2336049Z ##[section]Finishing: Detect Destroy Actions

**Agent Configuration**
 - OS: linux - ubuntu
 - Self Hosted
 - 0.13.4
 - AzureCLI version used (Default for hosted agent is acceptable or N/A)

**Additional context**
-

--- a/tasks/terraform-cli/src/commands/tf-show.ts
+++ b/tasks/terraform-cli/src/commands/tf-show.ts
@@ -44,14 +44,17 @@ export class TerraformShow implements ICommand {
         let jsonResult = JSON.parse(resultNoEol);
         const deleteValue = "delete";
 
-        for (let resourceChange  of jsonResult.resource_changes) {
-            if  (resourceChange.change.actions.includes(deleteValue))
-            {
-                this.setDestroyDetectedFlag(ctx, true);
-                this.logger.warning("Destroy detected!")
-                return;
+        if(jsonResult.resource_changes){
+            for (let resourceChange  of jsonResult.resource_changes) {
+                if  (resourceChange.change.actions.includes(deleteValue))
+                {
+                    this.setDestroyDetectedFlag(ctx, true);
+                    this.logger.warning("Destroy detected!")
+                    return;
+                }
             }
         }
+        
         this.logger.debug("No destroy detected")
         this.setDestroyDetectedFlag(ctx, false);
     }

--- a/tasks/terraform-cli/src/tests/features/terraform-show.feature
+++ b/tasks/terraform-cli/src/tests/features/terraform-show.feature
@@ -81,3 +81,14 @@ Feature: terraform show
         And the terraform cli task fails with message "terraform show command supports only env files, no tfvars are allowed during this stage."
         And pipeline variable "TERRAFORM_LAST_EXITCODE" is set to "1"
         And pipeline variable "TERRAFORM_PLAN_HAS_DESTROY_CHANGES" is not set
+
+        Scenario: show tf plan file with output only
+        Given terraform exists
+        And terraform command is "show"
+        And running command "terraform show -json show.plan" returns successful result with stdout from file "./src/tests/stdout_tf_show_tfplan_output_only.json"
+        And the target plan or state file is "show.plan"
+        When the terraform cli task is run
+        Then the terraform cli task executed command "terraform show -json show.plan"
+        And the terraform cli task is successful
+        And pipeline variable "TERRAFORM_LAST_EXITCODE" is set to "0"
+        And pipeline variable "TERRAFORM_PLAN_HAS_DESTROY_CHANGES" is set to "false"

--- a/tasks/terraform-cli/src/tests/stdout_tf_show_tfplan_output_only.json
+++ b/tasks/terraform-cli/src/tests/stdout_tf_show_tfplan_output_only.json
@@ -1,0 +1,23 @@
+{
+    "format_version": "0.1",
+    "terraform_version": "0.13.4",
+    "planned_values": {
+        "outputs": {
+            "some_string": {
+                "sensitive": false,
+                "value": "NOT_EXISTS"
+            }
+        },
+        "root_module": {}
+    },
+    "output_changes": {
+        "some_string": {
+            "actions": [
+                "create"
+            ],
+            "before": null,
+            "after": "NOT_EXISTS",
+            "after_unknown": false
+        }
+    }
+}

